packages/generators/add-generator.ts

```diff
@@ -79,29 +84,39 @@ const searchProps = (answers, input) => {
  *
  */
 
-module.exports = class AddGenerator extends Generator {
+export default class AddGenerator extends Generator {
+	private dependencies: string[];
+	private configuration: {
+		config: {
+			configName?: string,
+			topScope?: string[],
+			item?: string;
+			webpackOptions?: any,
+		},
+	};
+
 	constructor(args, opts) {
 		super(args, opts);
 		this.dependencies = [];
 		this.configuration = {
 			config: {
+				topScope: ["const webpack = require('webpack')"],
 				webpackOptions: {},
-				topScope: ["const webpack = require('webpack')"]
-			}
+			},
 		};
 		const { registerPrompt } = this.env.adapter.promptModule;
 		registerPrompt("autocomplete", inquirerAutoComplete);
 	}
 
-	prompting() {
-		let done = this.async();
-		let action;
-		let self = this;
-		let manualOrListInput = action =>
-			Input("actionAnswer", `What do you want to add to ${action}?`);
+	public prompting() {
+		const done: any = this.async();
+		let action: string;
+		const self: this = this;
+		let manualOrListInput: any = (promptAction: string) =>
+			Input("actionAnswer", `What do you want to add to ${promptAction}?`);
 
 		// first index indicates if it has a deep prop, 2nd indicates what kind of
-		let isDeepProp = [false, false];
+		const isDeepProp: any[] = [false, false];
```

Comment:
a bit naive to use `any` here when you can see the type

Reply:
Yeah. I tried `boolean[]` and `Array<object | boolean>` but both won't work because we also assign the values as boolean, string, object, array in different scopes. Some of the examples:

https://github.com/webpack/webpack-cli/blob/51dc005a800cd25e19f986adc6ff46436cc425b4/packages/generators/add-generator.js#L412

https://github.com/webpack/webpack-cli/blob/51dc005a800cd25e19f986adc6ff46436cc425b4/packages/generators/add-generator.js#L443

Reply:
https://stackoverflow.com/questions/40914716/dynamic-typescript-typing

Reply:
```js
const isDeepProp: [boolean, string | boolean | object | object[]]
```

^This is a valid type declaration which allows to control specifically which element should be of which type. Even after declaring the second element to be one of the boolean, string, object or array of objects, the compiler fails. I think the complex reassignment logic we're doing inside the generator is making the compiler's type prediction to fail.

TS compiler works for simple logic like doing `typeof var === "string"` and then using it as string, the compiler infers it to be a `string` even if the type was `string | boolean ...other types`.

Reply:
maybe add a interface for this if it's possible